src/mmda/predictors/heuristic_predictors/figure_table_predictors.py

```diff
@@ -110,58 +351,58 @@ def _predict(doc: Document, caption_type: str = 'Figure') -> List[SpanGroup]:
-                                                                        merged_boxes_caption_dict)
-        else:
-            merged_boxes_caption_dict = FigureTablePredictions.merge_spans(doc.vila_span_groups, caption_content='tab')
-            merged_boxes_fig_dict = FigureTablePredictions._merge_boxes(doc.layoutparser_span_groups,
-                                                                        merged_boxes_caption_dict, types=['Table'])
-
         predictions = []
-        for page in range(len(tqdm(doc.images))):
-            if merged_boxes_caption_dict.get(page) and merged_boxes_fig_dict.get(page):
+        predictions_captions = []
+        predictions_relations = []
+        for page in range(len(tqdm(self.doc.pages))):
+            if merged_boxes_caption_dict.get(page) and merged_boxes_fig_tab_dict.get(page):
                 cost_matrix = np.zeros(
-                    (len(merged_boxes_fig_dict[page]),
+                    (len(merged_boxes_fig_tab_dict[page]),
                      len(merged_boxes_caption_dict[page])))
-                for j, fig_box in enumerate(merged_boxes_fig_dict[page]):
-                    for i, caption_box in enumerate(merged_boxes_caption_dict[page]):
+                for j, fig_box in enumerate(merged_boxes_fig_tab_dict[page]):
+                    for i, span_group in enumerate(merged_boxes_caption_dict[page]):
+                        caption_box = span_group.box
                         assert hasattr(fig_box, 'box')
-                        assert hasattr(caption_box, 'box')
                         distance = FigureTablePredictions._get_object_caption_distance(
-                            fig_box.box, caption_box.box)
+                            fig_box.box, caption_box)
 
                         cost_matrix[j][i] = distance
                         if caption_type == 'Figure':
                             cost_matrix[j][i] = distance if distance > 0 else 900
 
                 row_ind, col_ind = linear_sum_assignment(cost_matrix)
                 for row, col in zip(row_ind, col_ind):
-                    predictions.append(SpanGroup(spans=[Span(
+                    span_group = SpanGroup(spans=[Span(
                         start=merged_boxes_caption_dict[page][col].start,
                         end=merged_boxes_caption_dict[page][col].end,
                         box=merged_boxes_caption_dict[page][col].box)],
-                        box_group=BoxGroup(boxes=[merged_boxes_fig_dict[page][row].box]),
+                        box_group=BoxGroup(boxes=[merged_boxes_fig_tab_dict[page][row].box]),
                         metadata=Metadata(type=caption_type)
-                    ))
-        return predictions
+                    )
+                    span_group.text = self.doc.symbols[span_group.start: span_group.end]
+                    if span_group.text.lower().startswith(caption_type.lower()[:3]):
+                        span_group.id = hash(json.dumps(span_group.to_json()))
```

Comment:
I think a simpler approach for generating the ids here is just to make them indices at time of creation. That's what we've done for other models that populate this field, and I think that's the mid-/long-term plan for MMDA, right @kyleclo ?

Reply:
For instance:

```python
box_group.id = len(predictions)
span_group.id = len(predictions_captions)
```

would probably be much cleaner and simpler.

Reply:
I like a lot more thank you, very neat, hopefully will not have collisions from different predictors span_groups.

Reply:
> For instance:
> 
> ```python
> box_group.id = len(predictions)
> span_group.id = len(predictions_captions)
> ```
> 
> would probably be much cleaner and simpler.

I agree as long as we do not need to have those id's unique. I have another use of the hash in the code, but in that case I do need to have ids unique. I am merging spang_groups which overlap and it is a graph based algorithm where I do use this ids and need to have them unique.

Reply:
The future for MMDA is that users will not be able to generate these ids themselves, they will automatically be assigned by MMDA, and they are only guaranteed to be unique within a set of spangroups (i.e. "vila_span_groups" or similar), not globally.

Since manual ID assignment -- especially assignment with the characteristics you are using here -- will eventually go away, I suggest you find another way to uniquely identify them. For instance, serial ID assignment as I suggested, prefixed with the name of the spangroup set would be sufficient, i.e. `f"figure_captions_${caption.id}"`